Azure/azure-cli
# [Output] Show update instruction in find, feedback and --help

**Description**  

To increase the discoverability of new versions, this PR adds the message to update Azure CLI in `az find`, `az feedback` and `az <command> --help`. The message is shown at most once in 7 days.
versions, update time and version check time is stored in `~/.azure/versionCheck.json`. Version data is valid when the update time is within 1 day.
Example value of the file:
```
{
    "update_time": "2020-05-14 17:07:10.291062",
    "versions": {
        "command-modules-nspkg": {
            "local": "2.0.3",
            "pypi": "2.0.3"
        },
        "dev-tools": {
            "local": "0.1.1"
        },
        "nspkg": {
            "local": "3.0.4",
            "pypi": "3.0.4"
        },
        "telemetry": {
            "local": "1.0.4",
            "pypi": "1.0.4"
        },
        "core": {
            "local": "2.5.1",
            "pypi": "2.5.1"
        },
        "azure-cli": {
            "local": "2.5.1",
            "pypi": "2.5.2"
        },
        "testsdk": {
            "local": "0.2.4"
        }
    },
    "check_time": "2020-05-14 17:08:14.105061"
}
```

The cache is used in the above commands, but not in `az --version`. With `az --version`, users always get the lastest versions from pypi.

The effect in `az group create --help`:
![image](https://user-images.githubusercontent.com/55177366/81134103-383af900-8f86-11ea-8fa7-63393deee1a3.png)

**Testing Guide**  

Create a new branch based on dev.
git pull https://github.com/fengzhou-msft/azure-cli.git update_instruction
Run the commands mentioned in Description.
You can manually modify/delete the cached files: `~/.azure/versionCheck.json` for testing purpose.
 

---

This checklist is used to make sure that common guidelines for a pull request are followed.
